tests/schema.cpp

```diff
@@ -450,6 +454,16 @@ TEST_CASE("Schema") {
             REQUIRE_THROWS_CONTAINING(schema.validate(), "Properties 'pk2' and 'pk1' are both marked as the primary key of 'object'.");
         }
 
+        SECTION("rejects primary key on embedded table") {
```

Comment:
We probably also need a migration test, e.g. if a local Realm adds or remove the embeddedness of a table during a migration.

Reply:
It is not possible to change embeddedness of a table after creation

Reply:
May I ask why not? For local Realms it just seems to be a boolean and some rules for transactions?

Reply:
Yes, it is just a flag, but the flag indicates that some invariants is upheld. We could probably check if a non embedded table complies to these invariants before making it embedded. This requirement was not captured in the tecnical design.

Reply:
Actually - the first implementation allowed this. And then we simplified. Apparently too much.